README.md

```diff
@@ -31,16 +31,27 @@ We recommend to install the following extensions to simplify the user experience
 - [Arm Tools Environment Manager](https://marketplace.visualstudio.com/items?itemName=Arm.environment-manager), an extension that allows to download, install, and manage software development tools using [Microsoft® Vcpkg](https://vcpkg.io/en/index.html) artifacts. Use this extension to for example install the `GCC compiler for ARM CPUs` which comes with a GDB variant for Arm CPUs.
 - [Arm CMSIS Solution](https://marketplace.visualstudio.com/items?itemName=Arm.cmsis-csolution), an extension that is a graphical user interface for csolution projects that use the [CMSIS-Toolbox](https://open-cmsis-pack.github.io/cmsis-toolbox/). Use this extension to build your csolution projects, to generate `*.cbuild-run.yml` debug configuration files, and to make use of contributed commands in your debug launch configurations.
 
-## pyOCD Debug Setup
+## Debug Setup
 
-- Install `GCC compiler for ARM CPUs` with the `Arm Tools Environment Manager` extension to get access to a GDB (`arm-none-eabi-gdb`).
+All debug setups require a GDB installation that supports the GDB remote protocol and can connect to a GDB server like pyOCD.
 
-## SEGGER® J-LINK® Debug Setup
+We recommend to install the `GCC compiler for ARM CPUs` using the `Arm Tools Environment Manager` extension.<br>
+It comes with `arm-none-eabi-gdb` which is used in the Arm CMSIS Debugger default debug configurations.
 
-- Install the latest [J-LINK Software and Documentation Pack](https://www.segger.com/downloads/jlink/#J-LinkSoftwareAndDocumentationPack) from [SEGGER](https://www.segger.com/). Ensure all required drivers and host platform specific settings are done.
-- Ensure the installation folder is added to your system's `PATH` environment variable. Alternatively, you can add an absolute path to your installation in the debug launch configuration.
+### pyOCD Debug Setup
 
-## Additional Extension Functionality
+This extension includes a pyOCD distribution which is used by default.
+
+Update your debug configuration's `target`>`server` setting to use a different pyOCD installation. Enter the full path to its `pyocd` executable (including the file name).
+
+### SEGGER® J-LINK® Debug Setup
+
+Install the latest [J-LINK Software and Documentation Pack](https://www.segger.com/downloads/jlink/#J-LinkSoftwareAndDocumentationPack) from [SEGGER](https://www.segger.com/). Ensure all required drivers and host platform specific settings are done.
+
+The extension expects the installation folder to be on your system `PATH` environment variable.<br>
+Alternatively, update your debug configuration's `target`>`server` setting to contain the full path to the J-LINK GDB server executable (including the file name).
+
+## Extension Functionality
 
 This extension contributes additional functionality to more seamlessly integrate the included extensions:
```

Comment:
This is a very awkward sentence! I think you mean "This extension contributes additional functionality to work seamlessly with other extensions".